frida_mode/test/fasan/GNUmakefile

```diff
@@ -28,6 +26,65 @@ else
 CFLAGS+=-Wno-pointer-arith
 endif
 
+ifndef ARCH
+
+ARCH=$(shell uname -m)
+ifeq "$(ARCH)" "aarch64"
+ ARCH:=arm64
+endif
+
+ifeq "$(ARCH)" "i686"
+ ARCH:=x86
+endif
+endif
+
+ifeq "$(ARCH)" "x86"
+LIBASAN_FILE:=libclang_rt.asan-i386.so
+endif
+
+ifeq "$(ARCH)" "x64"
+LIBASAN_FILE:=libclang_rt.asan-x86_64.so
+endif
+
+ifeq "$(ARCH)" "aarch64"
+LIBASAN_FILE:=libclang_rt.asan-aarch64.so
+endif
+
+# LIBASAN:=/usr/lib/llvm-10/lib/clang/10.0.0/lib/linux/libclang_rt.asan-x86_64.so
+# LIBASAN:=/usr/lib/x86_64-linux-gnu/libasan.so.6.0.0
+
+LLVM_CONFIG ?= llvm-config
+ifeq "$(shell test -e '$(shell which $(LLVM_CONFIG))' && echo 1)" "1"
+  $(info Found llvm-config: '$(shell which $(LLVM_CONFIG))')
+else
+  $(warning Cannot find llvm-config)
```

Comment:
isnt it better to $(errror ...) here as without llvm-config it wont work anyway? or is fasan just not available but everything else will be fine?

Reply:
ah you still try to work with clang below to find the lib ... this will only work for linux though, not MacOS or Android. but well ...

Reply:
The user can pass in the direct path to the library too if they like.

Reply:
The use can set `LIBASAN` manually: e.g. `LIBASAN=/path/to/my/dso make` So all is not lost at that point.